admission-webhook/utils_test.go

```diff
@@ -70,10 +77,21 @@ func buildPod(serviceAccountName string, podWindowsOptions *corev1.WindowsSecuri
 	}
 
 	shuffleContainers(containers)
-	podSpec := corev1.PodSpec{
-		ServiceAccountName: serviceAccountName,
-		Containers:         containers,
+
+	var podSpec corev1.PodSpec
+	if hostname != nil {
+		podSpec = corev1.PodSpec{
+			ServiceAccountName: serviceAccountName,
+			Containers:         containers,
+			Hostname:           *hostname,
+		}
+	} else {
+		podSpec = corev1.PodSpec{
+			ServiceAccountName: serviceAccountName,
+			Containers:         containers,
+		}
```

Comment:
nit: This could be simplified:

```suggestion
podSpec = corev1.PodSpec{
	ServiceAccountName: serviceAccountName,
	Containers:         containers,
}

if hostname != nil {
		podSpec.Hostname       *hostname,
		}
	} 
```

Reply:
will update